IncompatibleClassChangeError caused by Timer.Context refactor

The following commit changed Timer.Context from a class to an interface:
https://github.com/dropwizard/metrics/commit/1bacf4d645803f6241aeb0a9365dc44d2c43f2cf#diff-7423b8a39b3ee1cb8ad641557972ab24

This is causing a unique runtime issue (even if not a bug per se). If a class was compiled with an older version of Metrics where context was a class but is run on a new version of Metric where Context is an interface, Java is trowing:

> Caused by: java.lang.IncompatibleClassChangeError: Found interface com.codahale.metrics.Timer$Context, but class was expected

This basically renders v4.0 fully incompatible with v3.x if a dependency is using the Timer.Context class.

--- a/metrics-core/src/main/java/com/codahale/metrics/Timer.java
+++ b/metrics-core/src/main/java/com/codahale/metrics/Timer.java
@@ -9,13 +9,21 @@ import java.util.function.Supplier;
  * throughput statistics via {@link Meter}.
  */
 public class Timer implements Metered, Sampling {
-
     /**
      * A timing context.
      *
      * @see Timer#time()
      */
-    public interface Context extends AutoCloseable {
+    public static class Context implements AutoCloseable {
+        private final Timer timer;
+        private final Clock clock;
+        private final long startTime;
+
+        private Context(Timer timer, Clock clock) {
+            this.timer = timer;
+            this.clock = clock;
+            this.startTime = clock.getTick();
+        }
 
         /**
          * Updates the timer with the difference between current and start time. Call to this method will
@@ -23,37 +31,18 @@ public class Timer implements Metered, Sampling {
          *
          * @return the elapsed time in nanoseconds
          */
-        long stop();
+        public long stop() {
+            final long elapsed = clock.getTick() - startTime;
+            timer.update(elapsed, TimeUnit.NANOSECONDS);
+            return elapsed;
+        }
 
         /**
          * Equivalent to calling {@link #stop()}.
          */
         @Override
-        default void close() {
+        public void close() {
             stop();
-        }
-    }
-
-    /**
-     * The default timing context.
-     */
-    private static class DefaultContext implements Context {
-
-        private final Timer timer;
-        private final Clock clock;
-        private final long startTime;
-
-        private DefaultContext(Timer timer, Clock clock) {
-            this.timer = timer;
-            this.clock = clock;
-            this.startTime = clock.getTick();
-        }
-
-        @Override
-        public long stop() {
-            final long elapsed = clock.getTick() - startTime;
-            timer.update(elapsed, TimeUnit.NANOSECONDS);
-            return elapsed;
         }
     }
 
@@ -158,7 +147,7 @@ public class Timer implements Metered, Sampling {
      * @see Context
      */
     public Context time() {
-        return new DefaultContext(this, clock);
+        return new Context(this, clock);
     }
 
     @Override

